packages/meditrak-server/src/tests/routes/authenticate/authenticate.test.js

```diff
@@ -0,0 +1,174 @@
+      code: 'KI',
+    });
+
+    const { entity: laosEntity } = await findOrCreateDummyCountryEntity(models, {
+      code: 'LA',
+    });
+
+    const { entity: demoEntity } = await findOrCreateDummyCountryEntity(models, {
+      code: 'DL',
+    });
+
+    // Create test users
+    userAccount = await findOrCreateDummyRecord(models.user, {
+      first_name: 'Ash',
+      last_name: 'Ketchum',
+      email: 'ash-ketchum@pokemon.org',
+      ...hashAndSaltPassword(userAccountPassword),
+      verified_email: VERIFIED,
+    });
+
+    simpleApiUserAccount = await findOrCreateDummyRecord(models.user, {
+      first_name: 'simple',
+      last_name: 'api-client',
+      email: 'simple-api-client@pokemon.org',
+      verified_email: VERIFIED,
+    });
+    await findOrCreateDummyRecord(models.apiClient, {
+      username: simpleApiUserAccount.email,
+      user_account_id: simpleApiUserAccount.id,
+      secret_key_hash: encryptPassword(simpleApiClientSecret, process.env.API_CLIENT_SALT),
+    });
+
+    multiPermissionApiUserAccount = await findOrCreateDummyRecord(models.user, {
+      first_name: 'multiPermission',
+      last_name: 'api-client',
+      email: 'multi-permission-api-client@pokemon.org',
+      verified_email: VERIFIED,
+    });
+    await findOrCreateDummyRecord(models.apiClient, {
+      username: multiPermissionApiUserAccount.email,
+      user_account_id: multiPermissionApiUserAccount.id,
+      secret_key_hash: encryptPassword(multiPermissionApiClientSecret, process.env.API_CLIENT_SALT),
+    });
+
+    // Give the test users some permissions
+
+    // Public Demo Land Permission
+    await findOrCreateDummyRecord(models.userEntityPermission, {
+      user_id: userAccount.id,
+      entity_id: demoEntity.id,
+      permission_group_id: publicPermissionGroup.id,
+    });
+    await findOrCreateDummyRecord(models.userEntityPermission, {
+      user_id: simpleApiUserAccount.id,
+      entity_id: demoEntity.id,
+      permission_group_id: publicPermissionGroup.id,
+    });
+    await findOrCreateDummyRecord(models.userEntityPermission, {
+      user_id: userAccount.id,
+      entity_id: demoEntity.id,
+      permission_group_id: publicPermissionGroup.id,
+    });
+
+    await findOrCreateDummyRecord(models.userEntityPermission, {
+      user_id: multiPermissionApiUserAccount.id,
+      entity_id: kiribatiEntity.id,
+      permission_group_id: publicPermissionGroup.id,
+    });
+
+    // Public Laos permission
+    await findOrCreateDummyRecord(models.userEntityPermission, {
+      user_id: userAccount.id,
+      entity_id: laosEntity.id,
+      permission_group_id: publicPermissionGroup.id,
+    });
+
+    await findOrCreateDummyRecord(models.userEntityPermission, {
+      user_id: multiPermissionApiUserAccount.id,
+      entity_id: laosEntity.id,
+      permission_group_id: publicPermissionGroup.id,
+    });
+
+    // Public all other country permissions
```

Comment:
The comments above are only sometimes present, and this one doesn't seem to relate to the block below, so perhaps they're stale comments?